### PR TITLE
CYF-ITP-South Africa Emmanuel| Module-User-Focused-Data Form controls| Week2 

### PR DESCRIPTION
**Pull request by EmmanuelSiziba**
## Learners, PR Template

Self checklist

- [ yes] I have committed my files one by one, on purpose, and for a reason
- [ yes] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [yes ] I have tested my changes
- [ yes] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [ yes] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: Form-Controls/index.html [hunk 1]
```diff
@@ -14,8 +14,48 @@ <h1>Product Pick</h1>
...
-    <!-- write your html here-->
-    <!-- try writing out the requirements first-->
+    <header> <h1>Product Pick</h1>
+    </header>
+    <main>
+      <div class="wrapper">
+        <form id="product-form">
+        <div>
+        <label for="name">Name</label>
+        <input type="text" id="name" name="name" placeholder="Your name should be 9-20 characters long">
```
Comment (SallyMcGrath) [inline Form-Controls/index.html:24]: Do the requirements say a name must be 9-20 characters long? 
https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

File: Form-Controls/index.html [hunk 2]
```diff
@@ -14,8 +14,48 @@ <h1>Product Pick</h1>
...
+    </div>
+    <div>
+      <label for="email">Email</label>
+      <input type="email" id="email" name="email" placeholder="Enter your email address">
```
Comment (SallyMcGrath) [inline Form-Controls/index.html:28]: How could you **require** this field be filled in?

File: Form-Controls/index.html [hunk 3]
```diff
@@ -14,8 +14,48 @@ <h1>Product Pick</h1>
...
+     <option value="L">L</option>
+     <option value="XL">XL</option>
+     <option value="XXL">XXL</option>
+    </select>
+    </div>
+  
+  
+    <div>
+      <label for="delivery-date">Please enter the delivery date (YYYY-MM-DD):</label>
+      <input type="date" id="delivery-date" required>
```
Comment (SallyMcGrath) [inline Form-Controls/index.html:53]: In the acceptance criteria, it asks for you to constrain the date. What attributes could you use to set a minimum and maximum date range?

**Review by SallyMcGrath — CHANGES_REQUESTED** [review]
Thanks for this @EmmanuelSiziba . Your work is deployed here https://deploy-preview-233--cyf-ufd.netlify.app/form-controls/

It looks like you have brought along your Wireframe work into this PR. Make a new branch, from main, for each piece of work. If you don't know how to clean up this PR, bring your work to class and get help fixing it up there.

**Comment by EmmanuelSiziba** [comment]
hi Sally I had a huddle sassion with my fellow trainees on Friday, and they said the same thing as you in the comments you made. They said I made a branch inside a branch, I had been trying to fix ever since. So you are suggesting that I delete this pull request and the form control project in these branch and create the correct branch

**Comment by SallyMcGrath** [comment]
> hi Sally I had a huddle sassion with my fellow trainees on Friday, and they said the same thing as you in the comments you made. They said I made a branch inside a branch, I had been trying to fix ever since. So you are suggesting that I delete this pull request and the form control project in these branch and create the correct branch?

Yeah, there's a bunch of ways to fix this, but for now as you're just learning git, do this
- copy your form control folder to your desktop
- checkout main
- branch from main feature/form-controls
- drag your files in from your desktop and commit them
- open your PR from your new branch

Make a note in your calendar to come back to this problem in 1 month's time. By then, you will be able to fix it using `git cherry-pick`  https://www.atlassian.com/git/tutorials/cherry-pick /

File: Form-Controls/index.html (hunk 1 above)
Comment (EmmanuelSiziba) [reply Form-Controls/index.html:24]: Alot of websites I visit usually have a name restriction and I just made an assumption following this statement from the read me file  "  What is the customer's name? I must collect this data, and validate it. But what is a valid name? I must decide something." 

I read the article and I will fix my mistake. Thank you for the article

File: Form-Controls/index.html (hunk 3 above)
Comment (EmmanuelSiziba) [reply Form-Controls/index.html:53]: min and max attributes..?

File: Form-Controls/index.html (hunk 2 above)
Comment (EmmanuelSiziba) [reply Form-Controls/index.html:28]: do I need to specify the characters in I need in my email box?

**Comment by EmmanuelSiziba** [comment]
> > hi Sally I had a huddle sassion with my fellow trainees on Friday, and they said the same thing as you in the comments you made. They said I made a branch inside a branch, I had been trying to fix ever since. So you are suggesting that I delete this pull request and the form control project in these branch and create the correct branch?
> 
> Yeah, there's a bunch of ways to fix this, but for now as you're just learning git, do this
> 
> * copy your form control folder to your desktop
> * checkout main
> * branch from main feature/form-controls
> * drag your files in from your desktop and commit them
> * open your PR from your new branch
> 
> Make a note in your calendar to come back to this problem in 1 month's time. By then, you will be able to fix it using `git cherry-pick` https://www.atlassian.com/git/tutorials/cherry-pick /

**Comment by EmmanuelSiziba** [comment]
I was able to open my pull request the right way, now am working on the changes yo  commented on  I will open it once I am done.

**Pull request closed without merging**